File: TestBankApp/app/src/main/java/com/example/bankapp/repository/LoginRepository.java
package com.example.bankapp.repository;

import com.example.bankapp.domain.UserContract;
import com.example.bankapp.helper.BaseCallback;
import com.example.bankapp.model.user.UserAccountModel;
import com.example.bankapp.remote.BuildApi;
import com.example.bankapp.remote.LoginAPI;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class LoginRepository extends BuildApi implements UserContract.IRepository {


    @Override
    public void login(final String userName, final String password, final BaseCallback<UserAccountModel> result) {
        super.getBuild(LoginAPI.class)
                .login(userName, password).enqueue(new Callback<UserAccountModel>() {
            @Override
            public void onResponse(Call<UserAccountModel> call, Response<UserAccountModel> response) {
                if (response.body().getError().getMessage() != null) {
                    result.onUnsuccessful(response.body().getError().getMessage());
                    return;
                }


                result.onSuccessful(response.body());

            }

            @Override
            public void onFailure(Call<UserAccountModel> call, Throwable t) {
                result.onUnsuccessful(t.getMessage());
            }
        });
    }


}

File: BankApp/app/src/main/java/com/lucianogiardino/bankapp/data/repository/LoginRepository.kt
package com.lucianogiardino.bankapp.data.repository

import com.lucianogiardino.bankapp.data.RetrofitClient
import com.lucianogiardino.bankapp.domain.model.LoginResponse
import com.lucianogiardino.bankapp.presentation.login.LoginContract
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class LoginRepository: LoginContract.Repository{
    override fun login(listener: LoginContract.UseCase.OnLoginUserResponse, username: String, password: String) {
        RetrofitClient.instance.login(username,password).enqueue(
            object: Callback<LoginResponse> {
                override fun onFailure(call: Call<LoginResponse>, t: Throwable) {
                    listener.onLoginResponseFailed("Falha ao tentar efetuar o login")
                }
                override fun onResponse(call: Call<LoginResponse>,
                                        response: Response<LoginResponse>
                ) {

                    var message: String? = response.body()?.error?.message
                    var userAccount = response.body()!!.userAccount
                    if(message != null){
                        listener.onLoginResponseFailed(message)
                    }else{
                        listener.onLoginResponseSuccess(userAccount)
                    }
                }
            }
        )

    }

}